Argo/Types/Decoded/Decoded.swift

```diff
@@ -37,14 +73,39 @@ public extension Decoded {
 }
 
 public extension Decoded {
+  /**
+    Convenience function for creating `.TypeMismatch` errors.
+
+    - parameter expected: The expected type
```

Comment:
Considering `expected` is a `String` that reflects a type, some guidance on what to put here might be helpful.